drift-client/src/main/java/com/facebook/drift/client/DriftClientFactory.java

```diff
@@ -124,7 +128,10 @@ public DriftClientFactory(
                 statHandler = new NullMethodInvocationStat();
             }
 
-            DriftMethodHandler handler = new DriftMethodHandler(metadata, method.getHeaderParameters(), invoker, method.isAsync(), addressSelector, retryPolicy, statHandler);
+            // Create a bounded executor with a pool size at 4x number of processors
+            ExecutorService coreExecutor = newCachedThreadPool(daemonThreadsNamed("retry-service-%s"));
```

Comment:
Currently it will create a new thread pool for every single Drift method. This is not desirable. Ideally it should be a single executor reused by all the Drift clients. Please have a look at how Netty event loop is initialized, and create the thread pool similarly.

Reply:
moved it outside the for loop. won't be per method anymore, but per client. How is this?